pkg/middleware/middleware_test.go

```
@@ -0,0 +1,73 @@
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func TestWithSecurityHeaders(t *testing.T) {
+	inner := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	})
+	handler := WithSecurityHeaders(inner)
+
+	req := httptest.NewRequest(http.MethodGet, "/", nil)
```

golangci-lint (2.12.2)

[error] net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext

(noctx)